src/charm.py

```diff
@@ -170,6 +191,10 @@ def _command(self) -> str:
 
             args.append(f"--web.external-url={external_url}")
 
+        # enable remote write if an instance of the relation exists
+        if self.model.relations["remote-write-server"]:
+            args.append("--enable-feature=remote-write-receiver")
```

Comment:
Why condition this in the first place? Why not enable it all the time regardless?

Reply:
Add `--storage.remote.*` to config.yaml?

Reply:
We discussed this and wanted it to be conditional. I am open to changing it back.
What is `--storage.remote.*`?

Reply:
Having it conditional has the drawback of downtime due to prometheus restart. What are the advantages of having it conditional?

Prometheus command line arguments:
```
      --storage.remote.flush-deadline=<duration>  
                                 How long to wait flushing sample on shutdown or config reload.
      --storage.remote.read-sample-limit=5e7  
                                 Maximum overall number of samples to return via the remote read interface, in a single query. 0 means no limit. This limit is ignored for streamed response
                                 types.
      --storage.remote.read-concurrent-limit=10  
                                 Maximum number of concurrent remote read calls. 0 means no limit.
      --storage.remote.read-max-bytes-in-frame=1048576  
                                 Maximum number of bytes in a single frame for streaming remote read response types before marshalling. Note that client might have limit on frame size as
                                 well. 1MB as recommended by protobuf by default.
```

Reply:
Some use cases may want to secure their deployments. Remote write opens a port into which you can feed any junk data you like as long as it is in openmetrics format. Keeping it conditional seems like the right choice here.

Reply:
Without a relation there is no ingress, so you're safe against malicious externals; with a relation you have an ingress so you're vulnerable again. So seems like conditional remote write has no added security.
Besides, aren't charm relations themselves quite promiscuous as it is?

Reply:
No the port is still open without an ingress relation. While I agree that it doesn't matter in practice. When it comes to security, people tend to care about "in theory"